libraries/resource.rb

```diff
@@ -24,7 +24,7 @@ def allow(val)
           bin = Chef::Resource::DpkgAutostart.new('bin_file', context)
           bin.action :create
           current_resources = context.resource_collection.all_resources
-          [bin, current_resources].flatten.each_with_index do |res, i|
+          [current_resources, bin].flatten.each_with_index do |res, i|
             context.resource_collection[i] = res
```

Comment:
I was looking to see if there was a pull request already to fix the deprecation warning* on `[]=` and it seems a good fit with this since you're swapping to append and the `insert()` is actually for appending ...

```
[current_resources, bin].flatten.each do |res|
  context.resource_collection.insert(res)
end
```

* http://www.rubydoc.info/github/opscode/chef/Chef/ResourceCollection#%5B%5D%3D-instance_method